### PR TITLE
Convert findType -> Library#findType.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/dart-lang/source_gen/issues/170.

### CODE REVIEW

File: lib/src/library.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/src/dart/resolver/scope.dart';
+
+/// A high-level wrapper API with common functionality for [LibraryElement].
+class Library {
+  final LibraryElement _element;
+
+  Namespace _namespace;
```
Comment (jakemac53) [inline lib/src/library.dart]: Consider making this a getter which lazily creates the namespace, which would clean up the `findType` code a fair bit I think.

File: lib/src/library.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/src/dart/resolver/scope.dart';
+
+/// A high-level wrapper API with common functionality for [LibraryElement].
+class Library {
```
Comment (kevmoo) [inline lib/src/library.dart]: The name bothers me a bit. Seems a bit too generic.

Maybe LibraryHelper? Wrapper?

Not a huge deal...just thinking...

File: lib/src/library.dart (hunk 1 above)
Comment (matanlurey) [inline lib/src/library.dart]: Done.

File: lib/src/library.dart (hunk 2 above)
Comment (matanlurey) [inline lib/src/library.dart]: Renamed to `LibraryReader`.

**Pull request merged**